### ISSUE TITLE
ODF should handle split lines

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1226]
We saw this in a test file, and the docs say it's allowed:
![screen shot 2017-04-26 at 9 48 44 am](https://cloud.githubusercontent.com/assets/99469/25435204/a3b60320-2a65-11e7-9689-7e84875185a6.png)

**Comment by dankelley** [comment]
Oh, hang on. It says "**may**". So maybe yes, maybe no.  That's going to be a bit odd to code for. I guess we can hope that the same rule will be followed throughout a file. Using the word "**shall**" or "**is**" would have been nice.

**Comment by dankelley** [comment]
I'm attaching a test file. The test code
```R
library(oce)
d <- read.odf("ctd.odf")
```
produces error message
```
Error in scan(file = file, what = what, sep = sep, quote = quote, dec = dec,  : 
  line 2 did not have 11 elements
In addition: Warning messages:
1: In ODFNames2oceNames(ODFnames, ODFunits, PARAMETER_HEADER = NULL,  :
  length of ODFnames and ODFunits should agree but they are 21 and 20. Padding with empty units
2: In ODFNames2oceNames(ODFnames, ODFunits, PARAMETER_HEADER = NULL,  :
  unable to interpret ODFunits[5]='ma', for item named 'oxygenCurrent'
3: In ODFNames2oceNames(ODFnames, ODFunits, PARAMETER_HEADER = NULL,  :
  unable to interpret ODFunits[9]='ma', for item named 'oxygenCurrent2'
```
(Note: I am not going to address the warnings, because it simply seems to be an error in the ODF file, for a unit that is certainly not expressed in a conventional way.)

[ctd_odf.txt](https://github.com/dankelley/oce/files/981139/ctd_odf.txt)

**Comment by dankelley** [comment]
This works now in `develop`. I put the test file into the test suite that is run by
```
cd oce/tests_private
make odf_03.out
```
so we can be assured that future code evolution will not break this new methodology. Also, the other odf tests in that directory continue to test more conventional files.

To make this work, I had to drop `read.table` and hand-code something like that using `scan`, followed by `as.matrix` (with `ncol` as determined from the header), and finally with `as.data.frame` and then `as.list`. This all seemed simple until I realized that some of my files have a column for time, and that's non-numeric. Hm. So I wrote code to check whether each element in the first created line of the data frame contained letters. With that information, I could coerce columns before making a list. I doubt that this is much slower than `read.table`, since the latter has to do this sort of thing anyway.
